### PR TITLE
Adds zone to {BaseBastionLaunchJob,SubmitBastionJob}.Config

### PR DESCRIPTION
**Pull request by ruomingp**
This allows `axlearn gcp launch ...` to print commands with `--zone=...`.

### CODE REVIEW

File: axlearn/cloud/gcp/jobs/bastion_vm.py [hunk 1]
```diff
@@ -317,11 +317,23 @@ def _execute(self):
...
     """
 
+    @config_class
+    class Config(BaseSubmitBastionJob.Config):
+        zone: Required[str] = REQUIRED
+
+    @classmethod
+    def from_flags(cls, fv: flags.FlagValues, **kwargs) -> Config:
+        cfg = super().from_flags(fv, **kwargs)
+        cfg.zone = fv.zone
```
Comment (markblee) [inline axlearn/cloud/gcp/jobs/bastion_vm.py:334]: I think we can remove this (`from_flags` defaults to setting configs that match flag names).

File: axlearn/cloud/gcp/jobs/launch.py [hunk 2]
```diff
@@ -207,6 +210,7 @@ def from_flags(cls, fv: flags.FlagValues, action: str, **kwargs) -> Config:
         # Default output_dir depends on the final value of --name.
         fv.set_default("output_dir", f"gs://{gcp_settings('ttl_bucket')}/axlearn/jobs/{fv.name}")
         cfg = super().from_flags(fv, **kwargs)
+        cfg.zone = fv.zone
```
Comment (markblee) [inline axlearn/cloud/gcp/jobs/launch.py:213]: Same here.

**Review by markblee — APPROVED** [review]

File: axlearn/cloud/gcp/jobs/bastion_vm.py (hunk 1 above)
Comment (ruomingp) [reply axlearn/cloud/gcp/jobs/bastion_vm.py]: Oh, that's neat. Thanks!

File: axlearn/cloud/gcp/jobs/launch.py (hunk 2 above)
Comment (ruomingp) [reply axlearn/cloud/gcp/jobs/launch.py]: Done.

File: axlearn/cloud/gcp/jobs/bastion_vm.py [hunk 3]
```diff
@@ -335,7 +341,8 @@ def _execute(self):
             "\nView bastion outputs with:\n"
             f"gsutil cat {os.path.join(self.bastion_dir, 'logs', cfg.job_name)}\n"
             "\nCheck job history with:\n"
-            f"{infer_cli_name()} gcp bastion history --name={cfg.name} --job_name={cfg.job_name}"
+            f"axlearn gcp bastion history --name={cfg.name} "
```
Comment (markblee) [inline axlearn/cloud/gcp/jobs/bastion_vm.py:344]: Was the `infer_cli_name()` -> `axlearn` change necessary for some reason?

**Review by markblee — APPROVED** [review]

File: axlearn/cloud/gcp/jobs/bastion_vm.py (hunk 3 above)
Comment (ruomingp) [reply axlearn/cloud/gcp/jobs/bastion_vm.py:344]: Yes, I noticed that it doesn't work for `ajax gcp bastion history ...`.

**Pull request merged**